pyensign/utils/queue.py

```diff
@@ -35,6 +36,33 @@ async def read_response(self):
         """
         return await self._response_queue.get()
 
+    async def requests(self):
+        """
+        Return requests in the queue as an async iterator. The iterator is only
+        exhausted when the queue is closed. If the queue is empty this method will
+        block until there is a request in the queue.
+        """
+        while True:
+            request = await self.read_request()
+            if request is None:
+                break
+            yield request
+
+    async def flush_requests(
+        self, timeout=timedelta(seconds=2.0), tick=timedelta(milliseconds=100)
+    ):
+        """
+        Block until the request queue is empty or the timeout is reached.
+        """
+        while not self._request_queue.empty():
+            await asyncio.sleep(tick.total_seconds())
+            if timeout is not None:
+                timeout -= tick
+                if timeout <= timedelta():
+                    raise asyncio.TimeoutError(
+                        "timeout exceeded before all requests were flushed"
+                    )
```

Comment:
It might be wise to delegate timeout handling to `asyncio` since you cannot guarantee clock time will match sleep and processing time with the tick rate in this method. Allowing `asyncio` to handle the timeout will push that calculation out to the event loop. I think this loop should be something like:

```python
while not self._request_queue.empty():
    await asyncio.sleep(tick.total_seconds())
# emit EmptyEvent -- not sure how to do this
```

Then have some other callback that is listening for either the `EmptyEvent` or the `Timeout`.